src/torch/4-encode-classification-train-and-test-images/encode-training-and-test-images.lua

```diff
@@ -59,7 +59,7 @@ function encodeData(type, encoder)
     print('Class ' .. cl .. ' with files ' .. #files[cl + 1])
     local input = loadData(encoder, files[cl + 1])
```

Comment:
Probably nitpicking here, but wouldn't `loadData(encoder, files[cl + 1]):double()` do, instead of assigning to one variable then overwriting it with a type cast?